source/index.ts

```diff
@@ -32,9 +35,9 @@ function validator(fullPath, relativePath) {
  * @param {validateCallback} next - the completion callback
  * @returns {void}
  */
-function validate(fullPath, next) {
-	const invalidPaths = []
-	readdir(fullPath, validator.bind(invalidPaths), function (err) {
+export default function validate(fullPath: string, next: validateCallback) {
+	const invalidPaths: any = []
```

Comment:
don't cheat with any, should be `Array<string>` most probably — yep: `invalidPaths?: string[]` as defined in the errback callback

also, we should be converting this from callback to a promise api